src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EndpointInfoSourceTests.cs

```diff
@@ -254,70 +256,89 @@ private static void VerifyConnection(AppRunner runner, IEndpointInfo endpointInf
         private sealed class ServerEndpointInfoCallback : IEndpointInfoSourceCallbacks
         {
             private readonly ITestOutputHelper _outputHelper;
-            private readonly List<(AppRunner Runner, TaskCompletionSource<IEndpointInfo> CompletionSource)> _addedEndpointInfoSources = new();
+            private readonly SemaphoreSlim _completionEntriesSemaphore = new(1);
```

Comment:
nit: Can you add a comment to indicate you are using a semaphore so that you can cancel the lock?